doc/manual.rst

```diff
@@ -1708,16 +1705,16 @@ point to and modify the same location in memory (also called `aliasing`:idx:).
 
 Nim distinguishes between `traced`:idx: and `untraced`:idx: references.
 Untraced references are also called *pointers*. Traced references point to
-objects of a garbage collected heap, untraced references point to
-manually allocated objects or to objects somewhere else in memory. Thus
-untraced references are *unsafe*. However for certain low-level operations
+objects of a garbage-collected heap, untraced references point to
+manually allocated objects or objects somewhere else in memory. Thus
+untraced references are *unsafe*. However, for certain low-level operations
 (accessing the hardware) untraced references are unavoidable.
 
 Traced references are declared with the **ref** keyword, untraced references
 are declared with the **ptr** keyword.  In general, a `ptr T` is implicitly
 convertible to the `pointer` type.
 
-An empty subscript ``[]`` notation can be used to derefer a reference,
+An empty subscript ``[]`` notation can be used to de-refer a reference,
```

Comment:
Not sure about this one - there's no such word as "derefer" in the first place, and "dereference" is a valid word, but "de-reference" isn't

Reply:
This one is a bit controversial: I decided to put the hyphen pretty much as when you would when the version without a hyphen doesn't exist. And as you say - and I tried to find some source regarding this issue - "dereference" is a common term, "derefer" doesn't exist. (As a matter of fact, I even tried this very sentence with Grammarly and it suggests a change to "de-refer", marking "derefer" as wrong)

<img width="260" alt="Screenshot 2020-10-28 at 06 28 42" src="https://user-images.githubusercontent.com/1265028/97396193-de890b00-18e6-11eb-9044-c073fe4ce46f.png">